src/plugins/mongodb/mongodb_instance_config.cpp

```diff
@@ -106,6 +108,49 @@ MongoDBInstanceConfig::MongoDBInstanceConfig(Configuration *config,
 		}
 	}
 
+	if (enabled_) {
+		std::string extra_args;
+		try {
+			extra_args = config->get_string(prefix + "args");
+		} catch (Exception &e) {}
+		if (! extra_args.empty()) {
+			wordexp_t p;
+			int wrv = wordexp(extra_args.c_str(), &p, WRDE_NOCMD | WRDE_UNDEF);
+			switch (wrv) {
+			case WRDE_BADCHAR:
+				throw Exception("%s: invalid character in args", name());
+			case WRDE_BADVAL:
+				throw Exception("%s: undefined variable referenced in args", name());
+			case WRDE_CMDSUB:
+				throw Exception("%s: running sub-commands has been disabled for args", name());
+			case WRDE_NOSPACE:
+				throw OutOfMemoryException("Cannot parse args");
+			case WRDE_SYNTAX:
+				throw Exception("%s: shell syntax error in args", name());
+			default: break;
```

Comment:
I think the more specific exception is far more helpful! I did this because in examples I actually encountered different kinds of errors which I otherwise would not be able to distinguish and would have to guess.

Reply:
Yes, of course keep the specific cases. But if `wrv != 0` but also none of the error codes that you specified, it will continue as if there was no error.

Reply:
Oh, that makes perfect sense. Didn't get it that way on the first comment. Will update.